src/widgets/auto-styler.js

```diff
@@ -0,0 +1,141 @@
+var cdb = require('cartodb.js');
+var CategoryColors = require('./category/category-colors');
+var AutoStyler = cdb.core.Model.extend({
+  initialize: function (dataviewModel) {
+    this.dataviewModel = dataviewModel;
+    this.colors = new CategoryColors();
+    this.vector = !!this.dataviewModel._dataProvider;
+  },
+
+  getStyle: function () {
+    var style;
+    var widgetType = this.dataviewModel.get('type');
+    if (widgetType === 'category') {
```

Comment:
If you follow this route... perhaps we can merge `CategoryAutoStyler` and `CategoryColors`?